File: back/src/main/java/br/eng/eliseu/presente/model/core/BaseReportFilter.java
package br.eng.eliseu.presente.model.core;

import lombok.Getter;
import lombok.Setter;

import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
public abstract class BaseReportFilter {

    private Integer page = 0;
    private Integer size = 10;

    private String expand;

    private String nomeRelatorio;

    private String nomeArquivo;

}

File: front/src/app/shared/model/core/base-report-filter.ts

export abstract class BaseReportFilter {
    page: number = 0;
    size: number = 10;
    order?: string[] = ['id,ASC'];

    totalItens?: number = 0;

    expand?: string | string[];

    nomeRelatorio?: string;

    nomeArquivo?: string;

    constructor(init?: Partial<BaseReportFilter>) {
        Object.assign(this, init);
    }
}
